src/cms/page_xliff_converter.py

```diff
@@ -362,76 +336,132 @@ def _get_page_translation_slug(title):
-                    page_translation.slug = PageXliffHelper._get_page_translation_slug(page_xliff.title)
-
-                elif page.languages:
-                    target_language = None
-                    for language in page.region.languages:
-                        if page_xliff.language_code == language.code:
-                            target_language = language
-                            break
-                    if target_language:
-                        slug = PageXliffHelper._get_page_translation_slug(page_xliff.title)
-                        source_page_translation = list(page.translations.all())[0]
-                        page_translation = PageTranslation.objects.create(
-                            slug=slug,
-                            title=page_xliff.title,
-                            text=page_xliff.text,
-                            status=source_page_translation.status,
-                            language=target_language,
-                            public=source_page_translation.public,
-                            page=page,
-                            creator=user
-                        )
-
-                if page_translation:
-                    page_translation.save()
-                    result = True
-        # pylint: disable=broad-except
-        except Exception:
-            pass
+    def save_page_translation(trans_fields, user):
+        """
+        Save XLIFF input to page translation
+
+        :param trans_fields: translated fields extracted from XLIFF file
+        :type trans_fields: dict
+        :param user: author of translations
+        :type user: django.contrib.auth.models.User
+        :return: failure/success for each page or overall failed
+        :rtype: list or bool
+        """
+        page = Page.objects.get(id=int(trans_fields["page_id"]))
+        tgt_lang = Language.objects.get(code=trans_fields["tgt_lang_code"])
+        tgt_trans = PageTranslation.objects.get_or_create(page=page, language=tgt_lang)[0]
+        if tgt_trans:
+            tgt_trans.title = trans_fields["title"]
+            tgt_trans.text = trans_fields["content"]
+            tgt_trans.slug = PageXliffHelper._get_page_translation_slug(tgt_trans.title)
+            tgt_trans.version = tgt_trans.version + 1
+        else:
+            if tgt_lang in page.region.languages:
+                slug = PageXliffHelper._get_page_translation_slug(trans_fields["title"])
+                src_trans = page.get_translation(trans_fields["src_lang_code"])
+                tgt_trans = PageTranslation.objects.create(
+                    slug=slug,
+                    title=trans_fields["title"],
+                    text=trans_fields["content"],
+                    status=src_trans.status,
+                    language=tgt_lang,
+                    public=src_trans.public,
+                    page=page,
+                    creator=user
+                )
```

Comment:
```suggestion
        tgt_trans.title = trans_fields["title"]
        tgt_trans.text = trans_fields["content"]
        tgt_trans.slug = PageXliffHelper._get_page_translation_slug(tgt_trans.title)
        tgt_trans.version = tgt_trans.version + 1
```
After `get_or_create()`, the `tgt_trans`-object should exist in any case, right?